### PR TITLE
Add standalone macOS agent-cli release

### PR DESCRIPTION
**Pull request by mpscholten**
## Summary

- add Apple Silicon/macOS 14+ bundle and archive flake outputs that statically link the Haskell dependency graph and package the remaining non-system dylibs, runtime tools, data, and licenses
- use a compiled Haskell helper to recursively discover and relocate Mach-O dependencies with collision-safe names, then ad-hoc sign the bundle and make `agent-cli` discover bundled resources and helpers when invoked directly or through a symlink
- publish a checksummed standalone archive from `v*` tags and document installation for users without Nix

## Notes

macOS system libraries and frameworks cannot be fully statically linked, so the artifact is a relocatable standalone archive rather than one literally all-static Mach-O. It contains no Nix-store load commands and does not require Nix at runtime. This initial release target is Apple Silicon, macOS 14 or newer, with ad-hoc signing rather than Developer ID notarization.

## Validation

- `nix build --accept-flake-config --no-link .#agent-cli-macos-archive`
- `printf ':quit\n' | nix develop -c cabal repl agent-cli:exe:agent-cli`
- Haskell relocator compilation with `-Wall`, Fourmolu formatting, padded-replacement edge cases, and a real 72-dylib FFmpeg relocation/audio smoke test
- extracted archive checksum and gzip verification
- clean-environment invocation through an external symlink
- managed PostgreSQL start, doctor, `select 42`, and stop
- FFmpeg audio generation, Bun evaluation, ripgrep, zstd, PostgreSQL/initdb, bundled-resource, arm64, code-signature, and no-Nix-load-command checks across 186 Mach-O files
- `actionlint`, Nix parse/flake evaluation, and `git diff --check`

### CODE REVIEW

File: nix/macos-bundle.nix [hunk 1]
```diff
@@ -0,0 +1,350 @@
...
+
+            licenses="$out/share/doc/haskell-agent/licenses"
+            mkdir -p "$licenses"
+            install -m 644 ${portableFfmpeg.src}/LICENSE.md \
+                "$licenses/FFmpeg-LICENSE.md"
+            install -m 644 ${portableFfmpeg.src}/COPYING.LGPLv2.1 \
+                "$licenses/FFmpeg-COPYING.LGPLv2.1"
+            install -m 644 ${bunLicense} "$licenses/Bun-LICENSE.md"
+            install -m 644 ${pkgs.postgresql_18.src}/COPYRIGHT \
+                "$licenses/PostgreSQL-COPYRIGHT"
```
[the remark this answers is not in the document]
Comment (mpscholten) [reply nix/macos-bundle.nix:137]: Verified against this repository’s pinned nixpkgs: `nix build nixpkgs#postgresql_18.src` produces an unpacked `...-source` directory, and `COPYRIGHT` exists at its root. The full `.#agent-cli-macos-archive` derivation builds successfully, and the extracted archive contains the copied PostgreSQL license text. This concern therefore does not apply to the pinned source derivation.
